### PR TITLE
feat: doctor checks for "set -e" in test scripts

### LINKED ISSUE #206: update doctor command to check for good test.sh
They should have this best practice to ensure that any failed command fails the script,
```
set -euo pipefail
```

### PR DESCRIPTION
**Pull request by turadg**
Closes #206

## tooling improvements

>As long as it passes CI it should be fine.

I also tested locally with,
- `yarn build` in the package
- packages/synthetic-chain/dist/cli/cli.js doctor

That led to errors in many test.sh files so I cleaned those up.

I tried adopting the `-e` across all scripts but CI failed and I had to revert. We'll have them going forward.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2145,6 +2164,22 @@ __metadata:
   languageName: node
   linkType: hard
 
+"glob@npm:^11.0.0":
```
Comment (dckc) [inline yarn.lock:2167]: I'm surprised we don't already have a glob dep...

oh... this is in agoric-3-proposals.

In agoric-sdk, we have a menagerie:

```console
$ grep 'glob ' yarn.lock 
    glob "8.0.3"
    glob "8.0.3"
    glob "8.0.3"
    glob "^8.0.1"
    glob "^7.1.6"
    is-glob "^4.0.3"
    glob "^7.1.6"
    fast-glob "^3.3.2"
    is-glob "^4.0.3"
    glob "^8.0.1"
    glob "^10.2.2"
    is-glob "~4.0.1"
    is-glob "^4.0.3"
    is-glob "^4.0.0"
    is-glob "^4.0.1"
    is-glob "^4.0.3"
    dir-glob "^3.0.1"
    fast-glob "^3.2.9"
    dir-glob "^3.0.1"
    fast-glob "^3.2.11"
    is-glob "^4.0.1"
    is-extglob "^2.1.1"
    glob "^7.1.4"
    glob "^8.0.1"
    fast-glob "3.2.7"
    glob "7.1.4"
    glob "^8.0.1"
    glob "^7.1.3"
    glob "^10.0.0"
    glob "^7.1.3"
    glob "^10.3.7"
    glob "^7.0.0"
    glob "^7.1.4"
    glob "^10.4.1"
    fast-glob "3"
```

File: proposals/29:upgrade-9/test.sh [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
```
Comment (dckc) [inline proposals/29:upgrade-9/test.sh:2]: docs say...

> "set -euo pipefail" is recommended.

you didn't want to risk compat problems?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ and then fulfill the remaining checklist items before merging.
 Closes #
 
 # adopt a passed proposal
+
```
Comment (dckc) [inline .github/pull_request_template.md:11]: is the prettier style set up to stick?

or was this just a 1-time?

**Review by dckc — APPROVED** [review]
works for me.

File: .github/pull_request_template.md (hunk 3 above)
Comment (turadg) [reply .github/pull_request_template.md:11]: one time, it's not linted

File: proposals/29:upgrade-9/test.sh (hunk 2 above)
Comment (turadg) [reply proposals/29:upgrade-9/test.sh:2]: yeah, but I should follow the advice. updated.

**Review by samsiegart — APPROVED** [review]

File: proposals/29:upgrade-9/test.sh (hunk 2 above)
Comment (turadg) [reply proposals/29:upgrade-9/test.sh]: I had to revert it to get to green. Something for another time.

**Pull request merged**